repos/system_upgrade/el7toel8/actors/unsupportedupgradecheck/actor.py

```diff
@@ -0,0 +1,63 @@
+import os
+import json
+
+from leapp.actors import Actor
+from leapp.models import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.utils.audit import get_connection
+from leapp import reporting
+
+
+class UnsupportedUpgradeCheck(Actor):
+    """
+    Checks enviroment variables and produces a warning report if the upgrade is unsupported.
+
+    Upgrade is unsupported if any LEAPP_DEVEL_* variable has been used.
+    """
+
+    name = 'unsupported_upgrade_check'
+    consumes = ()
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        env = os.environ
+        debug_vars = {k: env[k] for k in env if k.startswith('LEAPP_DEVEL_')}
+        if debug_vars:
+            reporting.create_report([
+                reporting.Title('Upgrade inhibited due to usage of development variables'),
+                reporting.Summary(
+                    'One or more environment variables in the form of LEAPP_DEVEL_* have been detected.\n'
+                    'These variables are for development purposes only and can interfere with the upgrade '
+                    'process in unexpected ways. As such, a successful and safe upgrade process cannot be '
+                    'guaranteed and the upgrade is unsupported.\n'
+                    'Found development variables:\n- ' + '\n- '.join(debug_vars) + '\n'
+                ),
+                reporting.Severity(reporting.Severity.HIGH),
```

Comment:
I am not sure where this inhibitor comes from. I am out of the loop on this, but according to the task description this should be a warning only, not an inhibitor.

Reply:
@drehak do not use `os.environ`. See `IPUConfig` - that means `self.configuration.leapp_env_vars`. - That means, you can just check whether variable are there or not. As well, when you require new env var to be there, modify the related actor.

Reply:
Alright, I'm getting them from there now.